src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,128 @@
+import styled from 'styled-components';
+import React, { ReactNode } from 'react';
+import { StyledBaseDiv, StyledBaseSpan } from '../../htmlElements';
+import { useTheme } from '../../context';
+import { SubcomponentPropsType, StyledSubcomponentType } from '../commonTypes';
+import Skeleton from '../Skeleton/Skeleton';
+
+export type AvatarProps = {
+  initials?: ReactNode;
+  imgURL?: string;
+  size?: number;
+  shape?: number;
+  isLoading?: boolean;
+  isError?: boolean;
+  hasImage?: boolean;
+
+  StyledAvatarContainer?: StyledSubcomponentType;
+  StyledAvatarText?: StyledSubcomponentType;
+  StyledAvatarImage?: StyledSubcomponentType;
+
+  avatarProps?: SubcomponentPropsType;
+
+  avatarTextRef?: React.RefObject<HTMLSpanElement>;
+  avatarImageRef?: React.RefObject<HTMLImageElement>;
+  avatarContainerRef?: React.RefObject<HTMLDivElement>;
+};
+
+export const AvatarContainer = styled(StyledBaseDiv)`
+  ${({ size, shape }: { size: number; shape: string }) => {
+    return `
+      display: flex;
+      border-radius: ${`${shape}%`};
+      background-color: #eaeaea; 
+      padding: 1rem;
+      width: ${`${size * 3}rem`};
```

Comment:
Instead of needing to calculate relative sizes, use "em" - the text can be .33em and the container can be 3em (and set the font size to 1em)